lib/declaration.ts

```diff
@@ -99,20 +99,30 @@ export default class DeclarationTranspiler extends base.TranspilerBase {
     return path.join('.');
   }
 
-  private isAnonymousInterface(node: ts.Node): boolean {
-    if (!ts.isInterfaceDeclaration(node)) return false;
-    let interfaceDecl = node as base.ExtendedInterfaceDeclaration;
-    // If we were able to associate a variable declaration with the interface definition then
-    // the interface isn't actually anonymous.
-    return !interfaceDecl.classLikeVariableDeclaration;
+  private isAnonymous(node: ts.Node): boolean {
+    if (ts.isInterfaceDeclaration(node)) {
+      const extendedInterfaceDecl = node as base.ExtendedInterfaceDeclaration;
+      // If we were able to associate a variable declaration with the interface definition then the
+      // interface isn't actually anonymous.
+      return !extendedInterfaceDecl.classLikeVariableDeclaration;
+    } else if (ts.isClassLike(node)) {
```

Comment:
maybe make this behavior opt in with a flag. E.g.
something like "--trust-js-types"

Reply:
Done.